src/rewrite.ts

```diff
@@ -0,0 +1,54 @@
+export class RewriteRule {
+  find: RegExp;
+  replace: string;
+
+  constructor(find: RegExp, replace: string) {
+    this.find = find;
+    this.replace = replace;
+  }
+
+  apply(value: string): string {
+    return value.replace(this.find, this.replace);
+  }
+}
+
+export class RewriteRules {
+  private rules: RewriteRule[];
+
+  constructor() {
+    this.rules = [];
+  }
+
+  appendRule(rule: RewriteRule): RewriteRules {
+    this.rules.push(rule);
+    return this;
+  }
+
+  apply<T>(value: T): T {
+    if (typeof value === "object" && value !== null) {
+      // If the object is an array, iterate through each element and call the function recursively
+      if (Array.isArray(value)) {
+        return (value.map((v) => this.apply(v)) as any) as T;
+      }
+
+      // If the object is not an array, create a new object with the same keys,
+      // and call the function recursively on each value
+      const oldObj = value as { [key: string]: any };
+      const newObj: { [key: string]: any } = {};
+      for (const key of Object.keys(oldObj)) {
+        const newKey = this.apply(key);
+        const newValue = this.apply(oldObj[key]);
+        newObj[newKey] = newValue;
+      }
+      return newObj as T;
+    } else if (typeof value === "string") {
+      let s = value as string;
+      for (const rule of this.rules) {
+        s = rule.apply(value);
+      }
+      return (s as any) as T;
+    } else {
+      return value;
```

Comment:
**Suggestion:** We could log a warning/error when reaching this block

Reply:
This branch is expected — it'll happen for numbers, booleans, nulls, etc.

Reply:
Ah yep that make sense - I hadn't realised the breadth of inputs going into this function